pkg/api/validation/events.go

```diff
@@ -25,13 +30,45 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	// TODO: There is no namespace required for node.
-	if event.InvolvedObject.Kind != "Node" &&
-		event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
+
+	// Todo: Events may reference 3rd party object, and we can't check whether the object is namespaced. Suppose them are namespaced. Do check if we can get the piece of information.
+	groupVersion := event.InvolvedObject.APIVersion
+	if groupVersion == "v1" || groupVersion == "extensions/v1beta1" {
```

Comment:
Perhaps add to the comment that this should apply to all groups served by this apiserver.